contracts/coordinator/src/msg.rs

```diff
@@ -41,6 +46,55 @@ pub enum ExecuteMsg {
     },
     #[permission(Specific(prover))]
     SetActiveVerifiers { verifiers: HashSet<String> },
+
+    #[permission(Any)]
+    InstantiateChainContracts {
+        deployment_name: String,
+        params: Box<DeploymentParams>,
+    },
+}
+
+#[cw_serde]
+pub enum DeploymentParams {
+    Manual {
+        gateway_code_id: u64,
+        gateway_label: String,
+        verifier_code_id: u64,
+        verifier_label: String,
+        verifier_msg: VerifierMsg,
+        prover_code_id: u64,
+        prover_label: String,
+        prover_msg: ProverMsg,
+    },
```

Comment:
depending of the future variants, this enum could have a lot of content, maybe extract all those fields into a struct? Something like this:

```suggestion
pub enum DeploymentParams {
    Manual(ManualDeploymentParams),
}
```

Reply:
you could abstract even further by creating a struct that holds code_id, label and params, then reuse that struct for gateway, verifier, prover and whatever other future contract we might want to instantiate